### PR TITLE
add promise signature to fastify ts definition file & update ts example

### PR DESCRIPTION
**Pull request by chanlito**

### CODE REVIEW

File: fastify.d.ts [hunk 1]
```diff
@@ -242,6 +242,13 @@ declare namespace fastify {
      */
     options(url: string, handler: RequestHandler<HttpRequest, HttpResponse>): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
 
+    /**
+     * Starts the server on the given port after all the plugins are loaded,
+     * internally waits for the .ready() event. Using promise.
+     */
+    listen(port: number): Promise<http.Server>
+    listen(port: number, hostname?: string): Promise<http.Server>
```
Comment (mcollina) [inline fastify.d.ts]: I don’t think we are returning anything here. Is <http.Server> needed at all?

Comment (chanlito) [reply fastify.d.ts]: I see callbacks' usage return http.Server so, I added it.

File: examples/typescript-server.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import * as cors from 'cors'
 import { createReadStream } from 'fs'
 import * as http from 'http'
 
-const server = fastify()
+const server = fastify<http.Server, http.IncomingMessage, http.ServerResponse>()
```
Comment (mcollina) [inline examples/typescript-server.ts]: this should not be needed.

File: examples/typescript-server.ts [hunk 3]
```diff
@@ -48,7 +48,11 @@ server.use(cors())
 server.get('/', opts, getHelloHandler)
 server.get('/stream', getStreamHandler)
 
-server.listen(3000, err => {
-  if (err) throw err
-  console.log(`server listening on ${server.server.address().port}`)
```
Comment (mcollina) [inline examples/typescript-server.ts]: I think we should keep the previous callback thingy as well. This code is used only to validate the types and it does not run.

File: fastify.d.ts (hunk 1 above)
Comment (chanlito) [reply fastify.d.ts]: Hmm, probably not needed. Maybe should return just `Promise<void>`.

Comment (mcollina) [reply fastify.d.ts]: listen returns undefined everywhere. So you might as well fix it.

File: examples/typescript-server.ts (hunk 2 above)
Comment (chanlito) [reply examples/typescript-server.ts]: But `server.server` is missing type without those generics.

**Pull request closed without merging**